conan/tools/microsoft/msbuild.py

```diff
@@ -10,7 +10,7 @@ def msbuild_verbosity_cmd_line_arg(conanfile):
 
 
 def msbuild_arch(arch):
-    return {'x86': 'x86',
+    return {'x86': 'Win32',
```

Comment:
Bug not manifested, I think, because the only place where this would have been used was defaulted:

[conan/tools/microsoft/layout.py](/conan-io/conan/blob/b6bcdc5f24bb9ab5e4ca8143a42637b1f010d471/conan/tools/microsoft/layout.py#L13):

```python
    if conanfile.settings.arch != "x86":
        arch = msbuild_arch(conanfile.settings.arch)
```

Reply:
I can have a look later to those broken tests, thanks!

Reply:
I have access to VS2019 but not VS2017 or VS2015.

Creating an empty C++ project in VS2019 creates `x86` and `x64` configurations by default.

I'm not (normally) a Windows developer so not sure if these defaults changed recently. If they did `msbuild_arch` behaviour would depend on the msbuild version :/

Reply:
https://github.com/MicrosoftDocs/visualstudio-docs/issues/5419#issuecomment-637858985:

> "Win32" is used for C++ projects and it means "x86." I think what's confusing here is that you really have both "project-level platforms" and "solution-level platforms" and the project platforms come from the language-specific project systems. C++ projects use "Win32" and "x64", but the solution platforms use 'x86' and 'x64', NOT Win32. But, when you choose 'x86' as the solution configuration, VS selects the Win32 project configuration, so it works. This could be explained in the topic.

And [later](https://github.com/MicrosoftDocs/visualstudio-docs/issues/5419#issuecomment-885815357):

> As best as I can tell, whether a solution has "Win32" or "x86" as a platform depends on what version of Visual Studio it was originally created with. My oldest solutions have "Win32" and newer ones have "x86".

So I'm not sure what's best to do here.

Reply:
Yes, x86 is the configuration created, but internally, for some things, VS is expecting the ``Win32``.
Some tests were breaking, because for the command line it expects the ``x86``, while in the .props file it expects the ``Win32``. 

I have merged develop and added the architectures, without trying to use the ``msbuild_arch`` as common, please check if it works for your case.

Reply:
Thanks! Works for me.

Sorry about the kerfuffle over reusing `msbuild_arch`...